typings/index.d.ts

```diff
@@ -3031,13 +3031,11 @@ declare module 'discord.js' {
 
   type PartialTypes = 'USER' | 'CHANNEL' | 'GUILD_MEMBER' | 'MESSAGE' | 'REACTION';
 
-  interface PartialUser extends Partialize<User, 'bot' | 'flags' | 'locale' | 'system' | 'tag' | 'username'> {
+  interface PartialUser extends Omit<Partialize<User, 'bot' | 'flags' | 'locale' | 'system'>, "deleted"> {
     bot: User['bot'];
     flags: User['flags'];
     locale: User['locale'];
     system: User['system'];
-    readonly tag: null;
-    username: null;
```

Comment:
no, if you look at the `partial` getter, https://github.com/discordjs/discord.js/blob/b0ab37ddc0614910e032ccf423816e106c3804e5/src/structures/User.js#L120 you can see the username has to not be a string for it to be partial, and the `tag` getter https://github.com/discordjs/discord.js/blob/b0ab37ddc0614910e032ccf423816e106c3804e5/src/structures/User.js#L200 it returns null if there is no username, and the `Partialize` type sets it to `string | null` https://github.com/discordjs/discord.js/blob/b0ab37ddc0614910e032ccf423816e106c3804e5/typings/index.d.ts#L2940

Reply:
Ah, didn't realize partial users could have more than just the ID set. Apparently `USER_UPDATE` always contains the full user, not partial updates. I've reverted that change.